concepts/modularization/README.md

```diff
@@ -0,0 +1,170 @@
+- deployment of the operator (Kubernetes manifests / helm chart)
+
+
+# Component submission process
+To submit a new component to Kyma you need to prepare a component descriptor (with all related resources). 
+
+Component validation should be automated by governance jobs that check different aspects like:
+- operator CRD validation (e.g. status format)
+- exposing metrics
+- proper logging format
+- ...
+
+Governance jobs should be owned by teams (no shared responsibility)
+
+Apart from technical quality, Kyma components should fulfill other standards (24/7 support, documenting micro-deliveries, etc)
+
+# Operator based component management
+
+`kyma-operator` is a meta operator responsible for installing and updating component operators. It is similar to [Operator Lifecycle Manger](https://olm.operatorframework.io/), but it can work in 2 modes: 
+- central mode (managing thousands of clusters) 
+- single cluster mode (local installation)
+
+## Regular (local) component operators
+Each Kyma component has to provide the operator that manages the component lifecycle (installing, updating, configuring, deleting). The component provider (team) has to prepare one custom resource (ComponentDescriptor) that specifies how to install the operator (deployment yaml) and how to configure it (operator CRD, default values). 
+The complexity of managing installation in many clusters or using configured release channels is handled by kyma-operator and component providers can ship regular kubernetes operators. Such regular operators are installed in the target cluster (where components should be installed). 
+
+## Central component operators
+`kyma-operator` can also work with other component operators deployed centrally. In this case, `kyma-operator` creates resources for component operators in the central cluster. The Reconciliation logic of the central component operator can find a remote cluster kubeconfig by convention or reference (e.g. annotation). 
+
+## Central vs local operator
+
+With Kyma 2.0 we moved component installation (`kyma-installer`) from the cluster to the central place (reconciler). Each component was installed by the dedicated reconciler from the Kyma control plane. For some components, it perfectly makes sense (they interact mainly with other central systems to enable some integrations), but for some components that interact only with internal cluster resources, such a setup is suboptimal. The local Kubernetes operator designed and implemented accordingly to the best practices should have minimal resource requirements (few MB of memory and few mCPU). Therefore for the cases where you don't need access to central services or external resources it is recommended to use a regular operator (not central). Such an operator is much easier to develop and maintain for the component team, as an operator lifecycle management is still handled centrally by `kyma-operator` (together with `manifest-operator`).
```

Comment:
```suggestion
With Kyma 2.0, we moved the component installation (`kyma-installer`) from the cluster to a central place (reconciler). Each component was installed by the dedicated reconciler from the Kyma control plane. For some components, it perfectly makes sense because they interact mainly with other central systems to enable some integrations. But for some components that interact only with internal cluster resources, such a setup is suboptimal. The local Kubernetes operator designed and implemented according to the best practices should have minimal resource requirements (few MB of memory and few mCPU). Therefore, for the cases where you don't need access to central services or external resources, it is better to use a regular operator than a central one. A regular operator is much easier to develop and maintain for the component team, because the operator lifecycle management is still handled centrally by `kyma-operator` (together with `manifest-operator`).
```